### PR TITLE
Fix null pointer exception

### PR DESCRIPTION
**Pull request by parkerpersaud**
Fix logging of null pointer exception

### CODE REVIEW

File: src/main/java/me/karakelley/http/Request.java [hunk 1]
```diff
@@ -82,10 +82,11 @@ private void setHeaders() {
 
   private void setRequestLine() {
     String requestLine;
-    String[] parsedLine;
+    String[] parsedLine = new String[2];
```
Comment (mikeknep) [inline src/main/java/me/karakelley/http/Request.java:85]: Can you explain why this change is necessary?

File: src/main/java/me/karakelley/http/Request.java [hunk 2]
```diff
@@ -82,10 +82,11 @@ private void setHeaders() {
...
     try {
-      requestLine = reader.readLine();
-      parsedLine = requestLine.replaceAll("\\s+", " ").split(" ");
+      if ((requestLine = reader.readLine()) != null) {
+        parsedLine = requestLine.replaceAll("\\s+", " ").split(" ");
```
Comment (mikeknep) [inline src/main/java/me/karakelley/http/Request.java:88]: Just noticed this—it's ancillary to this PR since this line is just an indentation change, but: is there a reason you're not just splitting on `\\s+`? The sub from that to `" "` seems superfluous.

File: src/test/java/me/karakelley/http/RequestTest.java [hunk 3]
```diff
@@ -97,6 +96,17 @@ void something() throws Exception {
     }
   }
 
+  @Test
+  void testNullPointerExceptionIsNotThrown() {
```
Comment (mikeknep) [inline src/test/java/me/karakelley/http/RequestTest.java:100]: I'd rather see this test define what _does_ happen in this situation, rather than just calling out one thing that doesn't happen, for a few reasons:
1. This high-level situation (a socket connection opening up but no data coming through) seems plausible (in fact, we've seen it happen when poking around the server already), and so this test could better serve as documentation of expected behavior. As-is, I don't know what should happen; I only know one thing that won't.
2. There are other app-crashing things that can happen beyond exceptions (see Error and Throwable), so I feel like this test doesn't sufficiently prove that your server won't blow up when this happens.

File: src/main/java/me/karakelley/http/Request.java (hunk 1 above)
Comment (parkerpersaud) [reply src/main/java/me/karakelley/http/Request.java:85]: Because of the first `if` block where it checks for null, since we may not hit that line and initialize `parsedLine` then it may not be initialized therefore it's initialized on line 85, outside of the `if` block.
